File: FastStudy/models/Milestone.py
from beanie import Document
from typing import Dict
from models.Challenge import TierName, TierRequirement
from models.Challenge import ProgressUnit


class Milestone(Document):
    title: str
    description: str
    tiers: Dict[TierName, TierRequirement]
    progress_unit: ProgressUnit

    class Settings:
        name = "milestones"

File: vue-project/src/models/Milestone.ts
export enum TierName {
    BRONZE = "bronze",
    SILVER = "silver",
    GOLD = "gold",
    PLATINUM = "platinum",
}

export enum ProgressUnit {
    HOURS = "hours",
    DAYS = "days",
    BLOCKS = "blocks",
}

export interface TierRequirement {
    value: number; // hours, days, or blocks depending on milestone type
    coins: number;
    claimed?: boolean;
}

export interface Milestone {
    id: string;
    title: string;
    description: string;
    tiers: Record<TierName, TierRequirement>;
    progressUnit: ProgressUnit;
}
